### DISCUSSION TITLE
JSON formatting and general trimestrip construction

### DISCUSSION THREAD

**Discussion by Cameronseitz** [nasa/openmct#4301] [Community Q&A] [answered]
I tried following the procedure on the User Guide to create timestrips. My issue is that the plan, once created, just looks like a singular block on my time strip once dragged over. (see picture) It doesn't make a grant-type chart like in the user guide. To see if it was the JSON formatting, I tried exporting the timeline from the demo website (https://openmct-demo.herokuapp.com/#/browse/demo:root/demo:3?view=layout&tc.mode=local&tc.timeSystem=utc&tc.startDelta=900000&tc.endDelta=0) as a JSON and then importing that same file into the OpenMCT running locally, but it still appears as a block. Additionally, in the demo website mentioned earlier, it is referred to as a "Timeline", but on my local instance of OpenMCT, there is no Timeline option, just Plans, and timestrips. Is this a version difference issue? Any help would be awesome.
![bad timestrip](https://user-images.githubusercontent.com/91702651/136402206-e9aecf62-7d01-42f9-af27-d35fc3aa288c.png)

**Comment by shefalijoshi** [comment]
Hi @Cameronseitz 

The Timeline on the demo is an older (unsupported) version.

To get sample JSON for the newer Plan, please see the JSON linked here: https://github.com/nasa/openmct/blob/master/src/plugins/plan/pluginSpec.js#L118
You will need to change time of your Open MCT view to match the timestamp of the JSON there.

**Reply by Cameronseitz** [reply]
Thank you. Does it start from line 118 and end in 135, or without the "Test Group" part from 119 to 134?

**Comment by shefalijoshi** [comment, marked as answer]
Hi @Cameronseitz ,

The JSON in your file should look something like below (I've changed the time stamps to a span between `2021-10-12 20:00:00.000Z` and `2021-10-12 22:00:00.000Z`

`{
    "TEST-GROUP": [
        {
            "name": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua",
            "start": 1634068800000,
            "end": 1634072400000,
            "type": "TEST-GROUP",
            "color": "fuchsia",
            "textColor": "black"
        },
        {
            "name": "Sed ut perspiciatis",
            "start": 1634070600000,
            "end": 1634074200000,
            "type": "TEST-GROUP",
            "color": "fuchsia",
            "textColor": "black"
        }
    ]
}
`

Resulting in a plan similar to:
![image](https://user-images.githubusercontent.com/5247066/137020287-e991d40f-ab1d-4307-9399-38248a509ac7.png)

**Reply by Cameronseitz** [reply]
Thank you!

**Reply by Cameronseitz** [reply]
Are there additional, optional parameters that can be included in the Plan JSON similar to how the demo plan has a start and duration parameter?

![image](https://user-images.githubusercontent.com/91702651/137233813-bf45e862-d474-4d8b-bf4e-6fa7eb0c3e90.png)

**Reply by jvigliotta** [reply]
Hi @Cameronseitz I haven't worked with the timestrip itself, but I wouldn't see any issues included additional properties if you need them. The only caveat being that during future development, there could be a conflict if more properties are added on our end. @shefalijoshi  feel free to add more information if you see fit.
